internal/agent/config.go

```diff
@@ -7,49 +7,87 @@ import (
+	Cacert string `yaml:"ca-cert"`
+	// GenerateCert is the path to the cert file which is generated as the result of enrollment
+	GeneratedCert string `yaml:"generated-cert"`
+	// DataDir is the directory where the device's data is stored
+	DataDir string `yaml:"data-dir"`
+	// ConfigDir is the directory where the device's configuration is stored
+	ConfigDir string `yaml:"config-dir"`
+	// EnrollmentCertFile is the path to the enrollment certificate
+	EnrollmentCertFile string `yaml:"enrollment-cert-file"`
+	// EnrollmentKeyFile is the path to the enrollment key
+	EnrollmentKeyFile string `yaml:"enrollment-key-file"`
+	// ManagementEndpoint is the address of the device management server
+	ManagementEndpoint string `yaml:"management-endpoint,omitempty"`
+	// EnrollmentEndpoint is the address of the device enrollment server
+	EnrollmentEndpoint string `yaml:"enrollment-endpoint,omitempty"`
 	// EnrollmentUIEndpoint is the address of the device enrollment UI
-	EnrollmentUIEndpoint string `json:"enrollmentUIEndpoint,omitempty"`
-
-	// CertDir is the directory where the device's certificates are stored
-	CertDir string `json:"certDir,omitempty"`
+	EnrollmentUIEndpoint string `yaml:"enrollment-ui-endpoint,omitempty"`
 	// TPMPath is the path to the TPM device
-	TPMPath string `json:"tpmPath,omitempty"`
-	// FetchSpecInterval is the interval between two reads of the remote device spec
-	FetchSpecInterval util.Duration `json:"fetchSpecInterval,omitempty"`
+	TPMPath string `yaml:"tpm-path,omitempty"`
+	// SpecFetchInterval is the interval between two reads of the remote device spec
+	SpecFetchInterval time.Duration `yaml:"spec-fetch-interval,omitempty"`
 	// StatusUpdateInterval is the interval between two status updates
-	StatusUpdateInterval util.Duration `json:"statusUpdateInterval,omitempty"`
+	StatusUpdateInterval time.Duration `yaml:"status-update-interval,omitempty"`
 	// LogPrefix is the log prefix used for testing
-	LogPrefix string `json:"logPrefix,omitempty"`
+	LogPrefix string `yaml:"log-prefix,omitempty"`
 
 	// testRootDir is the root directory of the test agent
 	testRootDir string
 	// enrollmentMetricsCallback is a callback to report metrics about the enrollment process.
-	enrollmentMetricsCallback func(operation string, duractionSeconds float64, err error)
+	enrollmentMetricsCallback func(operation string, durationSeconds float64, err error)
 }
 
 func NewDefault() *Config {
 	return &Config{
-		ManagementServerEndpoint: "https://localhost:3333",
-		StatusUpdateInterval:     util.Duration(DefaultStatusUpdateInterval),
-		FetchSpecInterval:        util.Duration(DefaultFetchSpecInterval),
+		ManagementEndpoint:   DefaultManagementEndpoint,
+		EnrollmentEndpoint:   DefaultManagementEndpoint,
+		EnrollmentUIEndpoint: DefaultManagementEndpoint,
+		ConfigDir:            DefaultConfigDir,
+		DataDir:              DefaultDataDir,
+		Cacert:               filepath.Join(DefaultDataDir, DefaultCertsDirName, CacertFile),
```

Comment:
This one must be on the config dir, not the data dir, since it's provided via image/configuration, not downloaded by the agent.